streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorContextImpl.java

```diff
@@ -109,7 +94,7 @@ public void logChange(final String storeName,
                           final long timestamp) {
         throwUnsupportedOperationExceptionIfStandby("logChange");
         // Sending null headers to changelog topics (KIP-244)
-        collector.send(
+        streamTask.recordCollector().send(
```

Comment:
I tried to reduce the number of unnecessary local variables that could potentially get out of sync

Reply:
Sure, if the plan is to ultimately remove `streamTask` from the context then it makes sense to just save the collector reference instead